src/main/java/org/peergos/Server.java

```diff
@@ -48,11 +48,7 @@ public Server() throws Exception {
 
         List<MultiAddress> swarmAddresses = config.addresses.getSwarmAddresses();
         int hostPort = swarmAddresses.get(0).getPort();
-        //Optional<Object> p2pProxyEnabled = config.getOptionalProperty("Experimental","P2pHttpProxy");
-        //Optional<Object> bloomFilterSize = config.getOptionalProperty("Datastore","BloomFilterSize");
-        //Optional<Object> proxyTarget = config.getOptionalProperty("Addresses","ProxyTarget");
-        byte[] privKey = config.identity.privKey;
-        HostBuilder builder = new HostBuilder().setIdentity(privKey).listenLocalhost(hostPort);
+        HostBuilder builder = new HostBuilder().setEd25519Identity(config.identity.privKeyProtobuf).listenLocalhost(hostPort);
```

Comment:
what happens if the config has an RSA identity?

Reply:
I experimented with RSA -  setPrivKey(RsaKt.generateRsaKeyPair(2048).getFirst());
with no change to set identity method, it all still worked...

Reply:
fixed